pkg/asset/asset.go

```diff
@@ -55,3 +56,39 @@ func PersistToFile(asset WritableAsset, directory string) error {
 	}
 	return nil
 }
+
+// DeleteAssetFromDisk removes all the files for asset from disk.
+func DeleteAssetFromDisk(asset WritableAsset, directory string) error {
+	for _, f := range asset.Files() {
+		path := filepath.Join(directory, f.Filename)
+		if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+			return errors.Wrap(err, "failed to remove file")
+		}
+
+		dir := filepath.Dir(path)
+		ok, err := isDirEmpty(dir)
+		if err != nil && !os.IsNotExist(err) {
+			return errors.Wrapf(err, "failed to read parent directory for file %q", f.Filename)
```

Comment:
The only errors that come out of `isDirEmpty` are from [`Open` (which returns a `*PathError`)][1] and [`Readdirnames`][2] (which doesn't say).  And I haven't been able to come up with a test that passes `Open` but fails `Readdirnames` to check ;).  But I think we can drop the `%q` filename here and, if we want to be extra cautious, move it down onto the `Readdirnames` handler.  At this level, we only need something like:

```go
return errors.Wrapf(err, "is directory empty", f.Filename)
```

[1]: https://golang.org/pkg/os/#Open
[2]: https://golang.org/pkg/os/#File.Readdirnames

Reply:
> But I think we can drop the %q filename here 
sure.